src/main/java/com/wavefront/sdk/grpc/WavefrontClientInterceptor.java

```diff
@@ -0,0 +1,451 @@
+      this.allTags = ImmutableMap.<String, String>builder().
+          put(CLUSTER_TAG_KEY, applicationTags.getCluster() == null ? NULL_TAG_VAL :
+              applicationTags.getCluster()).
+          put(SERVICE_TAG_KEY, applicationTags.getService()).
+          put(SHARD_TAG_KEY, applicationTags.getShard() == null ? NULL_TAG_VAL :
+              applicationTags.getShard()).
+          put(GRPC_SERVICE_TAG_KEY, grpcService).
+          build();
+      this.histogramAllTags = ImmutableMap.<String, String>builder().
+          put(CLUSTER_TAG_KEY, applicationTags.getCluster() == null ? NULL_TAG_VAL :
+              applicationTags.getCluster()).
+          put(SERVICE_TAG_KEY, applicationTags.getService()).
+          put(SHARD_TAG_KEY, applicationTags.getShard() == null ? NULL_TAG_VAL :
+              applicationTags.getShard()).
+          put(GRPC_SERVICE_TAG_KEY, grpcService).
+          put(GRPC_METHOD_TAG_KEY, methodName).
+          build();
+      this.overallAggregatedPerSourceTags = ImmutableMap.<String, String>builder().
+          put(CLUSTER_TAG_KEY, applicationTags.getCluster() == null ? NULL_TAG_VAL :
+              applicationTags.getCluster()).
+          put(SERVICE_TAG_KEY, applicationTags.getService()).
+          put(SHARD_TAG_KEY, applicationTags.getShard() == null ? NULL_TAG_VAL :
+              applicationTags.getShard()).
+          build();
+      // update requests inflight
+      getGaugeValue(new MetricName(REQUEST_PREFIX + methodName + ".inflight", allTags)).
+          incrementAndGet();
+      getGaugeValue(new MetricName("client.total_requests.inflight",
+          overallAggregatedPerSourceTags)).incrementAndGet();
+    }
+
+    @Override
+    public ClientStreamTracer newClientStreamTracer(CallOptions callOptions, Metadata headers) {
+      if (span != null) {
+        tracer.inject(span.context(), Format.Builtin.HTTP_HEADERS, new TextMap() {
+          @Override
+          public void put(String key, String value) {
+            Metadata.Key<String> headerKey = Metadata.Key.of(key, Metadata.ASCII_STRING_MARSHALLER);
+            headers.put(headerKey, value);
+          }
+
+          @Override
+          public Iterator<Map.Entry<String, String>> iterator() {
+            throw new UnsupportedOperationException(
+                "TextMap should only be used with Tracer.inject()");
+          }
+        });
+      }
+      return new ClientTracer(this);
+    }
+
+    public void callEnded(Status status) {
+      if (streamClosed.getAndSet(true)) {
+        return;
+      }
+      long rpcLatency = System.currentTimeMillis() - startTime;
+      finishClientSpan(status);
+      String methodWithStatus = methodName + "." + status.getCode().toString();
+      // update requests inflight
+      getGaugeValue(new MetricName(REQUEST_PREFIX + methodName + ".inflight", allTags)).
+          decrementAndGet();
+      getGaugeValue(new MetricName("client.total_requests.inflight",
```

Comment:
Can we make this a constant. This metric name seems to be used in multiple places.

Reply:
yes, fixed.